perms/permissions_test.go

```diff
@@ -0,0 +1,81 @@
+package perms
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+)
+
+// TestMatchRegexURI tests the behaviour of the MatchRegexURI method of the
+// Manager.
+func TestMatchRegexURI(t *testing.T) {
+	// Construct a new Manager with a predefined list of perms.
+	m := &Manager{
+		perms: map[string][]bakery.Op{
+			"/lnrpc.WalletUnlocker/GenSeed":    {},
+			"/lnrpc.WalletUnlocker/InitWallet": {},
+			"/lnrpc.Lightning/SendCoins": {{
+				Entity: "onchain",
+				Action: "write",
+			}},
+			"/litrpc.Sessions/AddSession": {{
+				Entity: "sessions",
+				Action: "write",
+			}},
+			"/litrpc.Sessions/ListSessions": {{
+				Entity: "sessions",
+				Action: "read",
+			}},
+			"/litrpc.Sessions/RevokeSession": {{
+				Entity: "sessions",
+				Action: "write",
+			}},
+		},
+	}
+
+	// Assert that a full URI is not considered a wild card.
+	uris, isRegex := m.MatchRegexURI("/litrpc.Sessions/RevokeSession")
+	require.False(t, isRegex)
+	require.Empty(t, uris)
+
+	// Assert that an invalid URI is also caught as such.
+	uris, isRegex = m.MatchRegexURI("***")
+	require.False(t, isRegex)
+	require.Nil(t, uris)
+
+	// Assert that the function correctly matches on a valid wild card for
+	// litrpc URIs.
+	uris, isRegex = m.MatchRegexURI("/litrpc.Sessions/*")
+	require.True(t, isRegex)
+	require.ElementsMatch(t, uris, []string{
+		"/litrpc.Sessions/AddSession",
+		"/litrpc.Sessions/ListSessions",
+		"/litrpc.Sessions/RevokeSession",
+	})
+
+	// Assert that the function correctly matches on a valid wild card for
+	// lnd URIs. First we check that we can specify that only the
+	// "WalletUnlocker" methods should be included.
+	uris, isRegex = m.MatchRegexURI("/lnrpc.WalletUnlocker/*")
+	require.True(t, isRegex)
+	require.ElementsMatch(t, uris, []string{
+		"/lnrpc.WalletUnlocker/GenSeed",
+		"/lnrpc.WalletUnlocker/InitWallet",
+	})
+
+	// Now we check that we can include all the `lnrpc` methods.
+	uris, isRegex = m.MatchRegexURI("/lnrpc\\..*")
+	require.True(t, isRegex)
+	require.ElementsMatch(t, uris, []string{
+		"/lnrpc.WalletUnlocker/GenSeed",
+		"/lnrpc.WalletUnlocker/InitWallet",
+		"/lnrpc.Lightning/SendCoins",
+	})
```

Comment:
This code works and the tests pass, but I don't think `m.MatchRegexURI("/lnrpc.WalletUnlocker/*")` is doing what you think it does. The regex `/lnrpc.WalletUnlocker/*` matches all of the following strings:

```
/lnrpc.WalletUnlocker/GenSeed
/lnrpc.WalletUnlocker/InitWallet
/lnrpc.WalletUnlocker
/lnrpc.WalletUnlocker///////////////////////////
```

Reply:
You could use the regex `m.MatchRegexURI("/lnrpc.WalletUnlocker")` and it would have the same effect

Reply:
sure - can update to be more clear :+1:

Reply:
updated to match on "/lnrpc.WalletUnlocker/.*" rather